### PR TITLE
feat: support autoclass v2.1

### LINKED ISSUE #2306: Add support for Autoclass v2.1
Add support for new AutoclassConfig metadata fields. Full details are available in this [internal doc](http://goto.google.com/ac2-1-sdk-request).

Please wait for approval from myself before merging/releasing so we can [coordinate](http://goto.google.com/gcs-sdk-ac2-1) around release dates.

### PR DESCRIPTION
**Pull request by ddelgrosso1**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/nodejs-storage/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #2306  🦕

### CODE REVIEW

**Comment by ddelgrosso1** [comment]
Tests will fail until feature is rolled out.

**Review by danielbankhead — APPROVED** [review]

File: samples/getAutoclass.js [hunk 1]
```diff
@@ -37,7 +37,7 @@ function main(bucketName = 'my-bucket') {
   async function getAutoclass() {
     const [metadata] = await storage.bucket(bucketName).getMetadata();
     console.log(
-      `Autoclass enabled is set to ${metadata.autoclass.enabled} for ${metadata.name} at ${metadata.autoclass.toggleTime}.`
+      `Autoclass terminal storage class is set to ${metadata.autoclass.terminalStorageClass} for ${metadata.name} at ${metadata.autoclass.terminalStorageClassUpdateTime}.`
```
Comment (cojenco) [inline samples/getAutoclass.js:40]: For the getAutoclass sample, I think it'd be nice to demonstrate all four fields in the Autoclass config. Something along the lines of
```
console.log( 
  `Autoclass is ${metadata.autoclass.enabled ? 'enabled' : 'disabled'} for ${metadata.name} at ${metadata.autoclass.toggleTime}.
   Autoclass terminal storage class is last updated to ${metadata.autoclass.terminalStorageClass} at ${metadata.autoclass.terminalStorageClassUpdateTime}.`
)
```

**Review by cojenco — APPROVED** [review]
LGTM, thanks for working on this. We plan to release this feature in the week of 11/03

File: samples/setAutoclass.js [hunk 2]
```diff
@@ -14,32 +14,41 @@
  * limitations under the License.
  */
 
-function main(bucketName = 'my-bucket', toggle = false) {
+function main(
+  bucketName = 'my-bucket',
+  toggle = false,
+  terminalStorageClass = 'NEARLINE'
```
Comment (cojenco) [inline samples/setAutoclass.js:20]:
```suggestion
  toggle = true
  terminalStorageClass = 'ARCHIVE'
```

Apologies for the canonical sample changes, let's update this sample to set toggle = true with terminalStorageClass = 'ARCHIVE'. This is more aligned with the CUJ for setting terminal storage class. Thank you!

File: samples/getAutoclass.js (hunk 1 above)
Comment (ddelgrosso1) [reply samples/getAutoclass.js]: Done

File: samples/setAutoclass.js (hunk 2 above)
Comment (ddelgrosso1) [reply samples/setAutoclass.js]: Done

Comment (cojenco) [reply samples/setAutoclass.js]: lgtm, thanks Denis!

**Comment by cojenco** [comment]
This is ready for merge and release now, thanks @ddelgrosso1

**Pull request merged**